resim/metrics/default_report_metrics.py

```diff
@@ -0,0 +1,729 @@
+            and batch.jobs_metrics_status not in fail_statuses
+        )
+
+    if len(batches) == 0:
+        raise ValueError("Can't compute totals on zero batches!")
+
+    batches_sorted = all(
+        batches[i].creation_timestamp < batches[i + 1].creation_timestamp
+        for i in range(len(batches) - 1)
+    )
+
+    if not batches_sorted:
+        raise ValueError("Batches must be sorted!")
+
+    totals = {
+        "Number of Batches": len(batches),
+        "Number of Jobs": len(job_to_metrics_map),
+        "Number of Passing Batches": sum(batch_is_success(b) for b in batches),
+        "Number of Failing Batches": sum(batch_is_failure(b) for b in batches),
+        "Failed Jobs on Most Recent Batch": status_counts.tail(1)[
+            ["FAIL_WARN", "FAIL_BLOCK", "ERROR"]
+        ]
+        .sum(axis=1)
+        .iloc[0],
+        "Passed Jobs on Most Recent Batch": status_counts.tail(1)["PASSED"].iloc[0],
+    }
+
+    totals_status: dict[str, mu.MetricStatus] = {}
+
+    for k, v in totals.items():
+        totals[k] = np.array([v], dtype=np.float64)
+        totals_status[k] = np.array([mu.MetricStatus.PASSED_METRIC_STATUS])
+
+    totals_data = rm.GroupedMetricsData(
+        name="Totals Summary", category_to_series=totals
+    )
+
+    totals_status_data = rm.GroupedMetricsData(
+        name="Totals Summary Statuses", category_to_series=totals_status
+    )
+
+    failure_def = mu.DoubleFailureDefinition(fails_below=None, fails_above=None)
+    (
+        writer.add_double_summary_metric("Totals")
+        .with_description("High-level totals")
+        .with_blocking(False)
+        .with_should_display(True)
+        .with_status(mu.MetricStatus.PASSED_METRIC_STATUS)
+        .with_importance(mu.MetricImportance.ZERO_IMPORTANCE)
+        .with_value_data(totals_data)
+        .with_status_data(totals_status_data)
+        .with_failure_definition(failure_def)
+    )
+
+
+def flaky_experiences_metric(
+    writer: ResimMetricsWriter,
+    *,
+    batches: list[Batch],
+    batch_to_jobs_map: dict[str, list[Job]],
+    job_to_metrics_map: dict[str, list[JobMetric]],
+    batch_to_metrics_map: dict[str, list[BatchMetric]],
+    scalar_batch_metrics_map: dict[str, dict[str, MetricProto]],
+) -> None:
+    # pylint: disable=unused-argument
+    """Creates metrics listing flaky experiences and metrics
+
+    Creates a plotly table describing the experiences that failed most in this
+    report, job metrics that failed most, and batch metrics that failed most.
+
+    Args:
+        writer: A metrics writer to write these metrics to.
+        batches: All batches for this report.
+        batch_to_jobs_map: All jobs for each batch.
+        job_to_metrics_map: All job metrics for each job.
+        batch_to_metrics_map: All batch metrics for each batch.
+        scalar_batch_metrics_map: All scalar batch metrics for each batch as protos.
+    """
+    fail_statuses = (MetricStatus.FAIL_BLOCK, MetricStatus.FAIL_WARN)
```

Comment:
Might be fun to have the table display the count of fail_block and fail_warn beside the total failures? i.e.

Exp | Failures | Blocking Failures | Warning Failures

Reply:
Same as above comment. How do we treat the case of (jobs fail block, batch fail warn) or vice versa? I'm guessing its blocking if either is?

Reply:
Oh actually, that logic doesn't apply here. I should have thought about it more carefully.

Reply:
Added these for each table!